### PR TITLE
Add `expand`, `collapse`, `toggle` actions to `amp-accordion`

### LINKED ISSUE #10826: Add `expand`, `collapse`, `toggle` actions to `amp-accordion`
[amp-accordion](https://www.ampproject.org/docs/reference/components/amp-accordion) is an [AMP Extended Component ](https://www.ampproject.org/docs/reference/components) that allows you to display collapsible and expandable content sections. Sections toggle between expanded/collapsed states as user taps on the header section.

You can see a demo of this component in action on [AMP By Example](https://ampbyexample.com/components/amp-accordion/) website.

### The Issue
Currently there is no way to programmatically `expand`, `collapse`, `toggle`  a section of `amp-accordion`. This is a useful feature and we like to expose it as an [AMP Action](https://github.com/ampproject/amphtml/blob/master/spec/amp-actions-and-events.md) on the component, for instance we like to support the following code:

```
<button on="myAccordion.collapse(section=s1)">Collapse section 1</button>
<button on="myAccordion.expand(section=s2)">Expand section 2</button>
<button on="myAccordion.toggle(section=s3)">Toggle section 3</button>

<amp-accordion id="myAccordion">
  <section id="s1" expanded>
    <h4>Section 1</h4>
    <p>Content.</p>
  </section>
  <section id="s2" >
    <h4>Section 2</h4>
    <p>Content.</p>
  </section>
  <section id="s3">
    <h4>Section 3</h4>
    <p>Content.</p>
  </section>
</amp-accordion>
```

### Implementation strategy
`amp-accordion` already has code that handles these actions but the code is tied to click event.

- [ ] Move the toggle logic out of `onHeaderPicked_` method and into a new private method `toggle_(section, opt_expand)`
- [ ] Register new actions named `expand`, `collapse`, `toggle` using `this.registerAction` 
- [ ] Have the actions code:
   - [ ] parse the arguments, 
   - [ ] find the `section` parameter which is the id of the section to act on
   - [ ] find the section element with that id and call the newly created `toggle_(section, opt_expand)` 
method.
   - [ ] If `section` parameter is not provided, apply to method to ALL sections (e.g. `on=myAccordion.expand()` will expand all sections)
- [ ] Add tests
- [ ] Update`amp-accordion.md` to document the new feature.
- [ ] (Optional), update the AmpByExample for amp-accordion to show-case the new features.

### Step by step
- [ ] Claim this issue by adding a comment below.  Please only claim this bug if you plan on starting work in the next day or so.
- [ ] If you aren't too familiar with Git/GitHub, see the [Getting Started End-to-End Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md) for [an intro to Git & GitHub,](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#intro-to-git-and-github) and [how to get a copy of the code](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#get-a-copy-of-the-amphtml-code).  You can also refer to the [Quick Start Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md) for the necessary setup steps with less explanation than the End-to-End guide.
- [ ] Follow the instructions for [building AMP](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server).
- [ ] [Create a Git branch](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#create-a-git-branch) for making your changes.
- [ ] Follow the **Implementation strategy** from above.
- [ ] [Commit your changes](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#edit-files-and-commit-them) frequently.
- [ ] [Push your changes to GitHub](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#push-your-changes-to-your-github-fork).
- [ ] [Create a Pull Request](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#send-a-pull-request-ie-request-a-code-review).  Mention `closes Issue #10826` in the description.
- [ ] Assign @aghassemi as the reviewer
- [ ] [Respond to your reviewer's comments](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#respond-to-pull-request-comments) (if any).

Once approved, your changes will be merged.  **⚡⚡⚡Congrats on making your first contribution to the AMP Project!⚡⚡⚡**  You'll be able to see it [live across the web soon](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md)!

Thanks, and we hope to see more contributions from you soon.

### Questions?

If you have questions ask in this issue or on your Pull Request (mentioning @aghassemi) or see the [How to get help](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#how-to-get-help) section of the Getting Started guide.

### PR DESCRIPTION
**Pull request by mianuddin**
- Closes #10826
- Adds `expand`, `collapse`, `toggle` actions to `amp-accordion`
- Adds unit tests for aforementioned actions
- Adds documentation for aforementioned actions

Reviewer: @aghassemi

### CODE REVIEW

**Comment by mianuddin** [comment]
Whoops, I'll correct the issues found by Travis CI

File: extensions/amp-accordion/0.1/amp-accordion.js [hunk 1]
```diff
@@ -162,20 +196,23 @@ class AmpAccordion extends AMP.BaseElement {
...
-    const section = header.parentElement;
+  toggle_(section, opt_forceExpand) {
     const sectionComponents = section.children;
+    const header = sectionComponents[0];
     const content = sectionComponents[1];
     const contentId = content.getAttribute('id');
     const isSectionClosedAfterClick = section.hasAttribute('expanded');
     this.mutateElement(() => {
-      if (section.hasAttribute('expanded')) {
+      if (typeof opt_forceExpand !== 'undefined' && opt_forceExpand) {
```
Comment (aghassemi) [inline extensions/amp-accordion/0.1/amp-accordion.js]: let's simplify this with something like

```
let toExpand;
if(opt_forceExpand === true) {
toExpand = true;
} else if( opt_forceExpand === false) {
toExpand = false;
} else {
toExpnd = section.hasAttribute('expanded')
}
if(toExpand) {
...
} else {
..
}


```

**Comment by aghassemi** [comment]
Thanks for the PR @mianuddin, left some comments. Nothing major.

**Comment by aghassemi** [comment]
@mianuddin, friendly ping regarding the review comments.

**Comment by mianuddin** [comment]
@aghassemi Sorry for not implementing the changes sooner, I just had a big midterm. I'm aiming to complete the changes by end of week.

**Comment by aghassemi** [comment]
@mianuddin no rush, take your time. Just wanted to make sure still interested in continuing the work.

**Comment by mianuddin** [comment]
@aghassemi Apologies for the late commits. I've made the changes requested and fixed as much as I could from Travis CI. Your comments were extremely helpful, I really appreciate it.

**Comment by aghassemi** [comment]
@mianuddin can you please rebase (or send another PR if too tangled to fix). PR shows 259 files changed and tons of unrelated commits

**Comment by mianuddin** [comment]
@aghassemi I was able to rebase it 👍

**Comment by aghassemi** [comment]
@mianuddin  just a few lint errors causing tests to fail.

**Comment by mianuddin** [comment]
@aghassemi I'll get on those, thanks!

**Comment by mianuddin** [comment]
@aghassemi You're good to go! 💯

**Pull request merged**